consensus/parlia/snapshot.go

```diff
@@ -283,7 +284,9 @@ func (s *Snapshot) apply(headers []*types.Header, chain consensus.ChainHeaderRea
 				}
 			}
 		}
-		snap.updateAttestation(header, chainConfig, s.config)
+		if chainConfig.IsLynn(header.Number) || parliaEngine.verifyVoteAttestation(chain, header, headers[:index]) == nil {
```

Comment:
==>  `chainConfig.IsLynn(header.Number) || (chainConfig.IsBoneh(header.Number) && parliaEngine.verifyVoteAttestation(chain, header, headers[:index]) == nil)`

Reply:
This is a bit tricky here. Please comment and explain what the logic exactly means here. 

TODO: A bit coupling here that we pass `parlia engine` down to snapshot, anyway to improve?

Reply:
> ==> `chainConfig.IsLynn(header.Number) || (chainConfig.IsBoneh(header.Number) && parliaEngine.verifyVoteAttestation(chain, header, headers[:index]) == nil)`

> verifyVoteAttestation and updateAttestation both have checked `IsBoneh`
So suggested code behavior the same as before,
but do checking here may increasing readability, so accepted

Reply:
> ==> `chainConfig.IsLynn(header.Number) || (chainConfig.IsBoneh(header.Number) && parliaEngine.verifyVoteAttestation(chain, header, headers[:index]) == nil)`

verifyVoteAttestation and updateAttestation both have checked IsBoneh
So suggested code behavior the same as before,
but do checking here may increasing readability, so accepted

Reply:
> This is a bit tricky here. Please comment and explain what the logic exactly means here.
> 
> TODO: A bit coupling here that we pass `parlia engine` down to snapshot, anyway to improve?


1. more comment, accepted.
2. I tried to move verifyVoteAttestation as a func of snapshot
      tried to pass some flag to func apply.
      pass `parlia engine` is the last choice I pick.

      I will try to dig more deeply to avoid above coupling